server_structs/director.go

```diff
@@ -145,8 +152,131 @@ type (
+}
+func (x *XPelNs) ParseRawResponse(resp *http.Response) error {
+	raw := resp.Header.Values(x.GetName())
+	if len(raw) == 0 {
+		return errors.Errorf("No %s header found.", x.GetName())
+	}
+	keyDict := utils.HeaderParser(raw[0])
+	x.Namespace = keyDict["namespace"]
+	x.RequireToken, _ = strconv.ParseBool(keyDict["require-token"])
+	if keyDict["collections-url"] != "" {
+		x.CollectionsUrl, _ = url.Parse(keyDict["collections-url"])
+	}
+	return nil
+}
+
+func (x XPelAuth) GetName() string {
+	return "X-Pelican-Authorization"
+}
+func (x *XPelAuth) ParseRawResponse(resp *http.Response) error {
+	// If the director provides an auth header, raw will have an array of length 1.
+	raw := resp.Header.Values(x.GetName())
+	if len(raw) > 0 {
+		x.Issuers = make([]*url.URL, 0)
+		// clean up the string and split it by commas to fetch each issuer. Can't use
+		// utils.HeaderParser, because we don't have unique keys here.
+		cleaned := strings.ReplaceAll(raw[0], " ", "")
+		issuers := strings.Split(cleaned, ",")
+		for _, issuer := range issuers {
+			issuerUrlStr := strings.TrimPrefix(issuer, "issuer=")
+			issuerUrl, err := url.Parse(issuerUrlStr)
+			if err != nil {
+				return errors.Errorf("Failed to parse issuer URL %s from Director's %s header: %v", issuerUrlStr, x.GetName(), err)
+			}
+			x.Issuers = append(x.Issuers, issuerUrl)
+		}
+	}
+	return nil
+}
+
+func (x XPelTokGen) GetName() string {
+	return "X-Pelican-Token-Generation"
+}
+func (x *XPelTokGen) ParseRawResponse(resp *http.Response) error {
+	raw := resp.Header.Values(x.GetName())
+	if len(raw) > 0 {
+		// Parse issuer, for now assuming a single value but eventually may be multiple
+		x.Issuers = make([]*url.URL, 0)
+		keyDict := utils.HeaderParser(raw[0])
+		issuerUrl, err := url.Parse(keyDict["issuer"])
+		if err != nil {
+			return errors.Errorf("Failed to parse issuer URL %s from Director's %s header: %v", keyDict["issuer"], x.GetName(), err)
+		}
+		x.Issuers = append(x.Issuers, issuerUrl)
+
+		// Parse scope depth
+		maxScopeDepth, err := strconv.ParseUint(keyDict["max-scope-depth"], 10, 32)
+		if err != nil {
+			return errors.Errorf("Failed to parse max-scope-depth %s from Director's %s header: %v", keyDict["max-scope-depth"], x.GetName(), err)
+		}
+		x.MaxScopeDepth = uint(maxScopeDepth)
+
+		// Parse strategy
+		strategy, exists := keyDict["strategy"]
+		if !exists {
+			return errors.Errorf("No credential generation strategy found in Director's %s header", x.GetName())
+		}
+		if !IsValidStrategy(strategy) {
+			return errors.Errorf("Invalid strategy '%s' from Director's %s header", strategy, x.GetName())
+		}
+		x.Strategy = StrategyType(strategy)
+
+		// Parse base path(s) -- Right now we assume a single value, although this may eventually change.
+		basePath, exists := keyDict["base-path"]
+		if exists {
+			x.BasePaths = append(x.BasePaths, basePath)
+		}
+
+		// Handle potential for vault server in header
+		vaultServer, exists := keyDict["vault-server"]
```

Comment:
I'm more unsure about this one. We technically don't add it to the Director headers at this point, but I assumed this is an upcoming part of the header because of its inclusion in the `TokenGen` struct from `NamespaceAdV2`, where it lives along side all of the other values we _do_ pass in the Director headers. What's the big picture here? I can either a) remove it from the header struct and make sure we don't check for it anywhere in the client, b) leave it here if we plan to turn this on at some point soon, or c) add it to the actual header we generate in the Director. Thoughts?

Reply:
I say we leave it in for now and then make a separate issue to add it to the actual header we generate in the Director.